### ISSUE TITLE
twisted.web.microdom.Node.isEqualToNode does no actual comparison

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2542]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)| @dreid reported|
|-|-|
|Trac ID|trac#2542|
|Type|defect|
|Created|2007-03-28 17:37:44Z|
|Branch|https://github.com/twisted/twisted/tree/isequaltonode-2542|

While this makes isEqualToNode very fast it doesn't make it correct.  More importantly other tests use it to assert that two nodes are equal.

Attachments:

* [testNodeEquality.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/905/9058ef171b8c63928e3da0af09a2d63eceb4d974/b6fce9bf45862bbde73b985ebcfb15fc64f35b31.diff) (946 bytes) - added by dreid on 2007-03-28 17:38:53Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__2542 2542
type__defect defect
reporter__dreid dreid
priority__normal normal
milestone__ 
branch__branches_isequaltonode_2542 branches/isequaltonode-2542
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1175103464000000 1175103464000000
changetime__1222114686000000 1222114686000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jyknight's avatar" src="https://avatars.githubusercontent.com/u/873903?s=50" width="50" height="50">](https://github.com/jyknight)<a name="note_1"></a>|@jyknight commented|
|-|-|

For future reference: it looks like dp broke it in r8087, with commit message:
> Major woven optimizations, see changelog; Applied patch from [#7483](https://github.com/twisted/twisted/issues/7483)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [24727]) Branching to 'isequaltonode-2542'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun removed owner|
|-|-|

Fixed I guess.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set owner to @exarkun|
|-|-|

Cool, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [24906]) Merge isequaltonode-2542

Author: exarkun
Reviewer: therve
Fixes: [#2542](https://github.com/twisted/twisted/issues/2542)

Fix the definition of `Node.isEqualToNode` and the equality
methods on all `Node` subclasses so that it does not return
True when the node is not equal to the other object passed
in.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
